test/constants_test.dart

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:build_test/build_test.dart';
+import 'package:source_gen/source_gen.dart';
+import 'package:test/test.dart';
+
+void main() {
+  group('Constant', () {
+    List<Constant> constants;
+
+    setUpAll(() async {
+      constants = (await resolveSource(r'''
```

Comment:
nit: assign the result of `resolveSource` to a local variable to make this a bit easier to follow. In general I try not to wrap await calls in parens and use local vars instead because it becomes hard to follow.